src/GitHub.Api/Metrics/UsageTracker.cs

```diff
@@ -79,13 +81,16 @@ private void RunTimer(int seconds)
 
             var usageStore = usageLoader.Load(userId);
 
-            if (usageStore.LastUpdated.Date != DateTimeOffset.UtcNow.Date)
+            var currentTimeOffset = DateTimeOffset.UtcNow;
+            if (usageStore.LastUpdated.Date == currentTimeOffset)
             {
-                var currentTimeOffset = DateTimeOffset.UtcNow;
-                var beforeDate = currentTimeOffset.Date;
+                return;
+            }
 
+            lock (_lock)
```

Comment:
This lock is not needed, we're not reading or writing to disk at this point.

Reply:
Oh, I see, this covers the usage save, I see what you're trying to do. I don't think it's a good idea to lock all of this, when PostUsage might take a really long time to execute and all the other threads trying to save metrics will be hanging. If people enter play mode while this is happening, all the blocked metrics saves will be lost. Locks needs to be as fast as possible going in and out.